### PR TITLE
[18.0][MIG] document_page: Migration to 18.0

### PR DESCRIPTION
**Pull request by natuan9**
**Depend on:**
- [[18.0][MIG] document_knowledge: Migration to 18.0](https://github.com/natuan9/knowledge/pull/1)

**OCA Port:**
- [[17.0][PW] document_page: Port from 14.0](https://github.com/natuan9/knowledge/pull/3/commits)

### CODE REVIEW

**Comment by thienvh332** [comment]
Hi @natuan9 ,
Looks like you haven't used oca-port to check for missing PRs from v14.0

Missing PRs need porting:
- `https://github.com/OCA/knowledge/pull/399`
- `https://github.com/OCA/knowledge/pull/474`

**Comment by natuan9** [comment]
Hi @thienvh332 , I've ported missing PR and updated description

**Comment by thienvh332** [comment]
Hi @natuan9 ,
Have you checked the content of this commit https://github.com/natuan9/knowledge/pull/2/commits/d1c36c4d922500c51efc11b1df21ddbf5a45e03c?

**Comment by natuan9** [comment]
@thienvh332 I've checked and migrated it

**Comment by thienvh332** [comment]
> @thienvh332 I've checked and migrated it

I mean are you sure it's working properly in v18?

**Comment by natuan9** [comment]
> I mean are you sure it's working properly in v18?

I tested before pushing. Do you see any errors?

**Comment by thienvh332** [comment]
I don't think it will work. Okay, I'll check it again.

**Comment by thienvh332** [comment]
Ok it seems the creator of that js file is not using it. But anyway please convert it to the new js style of odoo 18.

Update the kanban view. Based on one of the following contexts:
- `https://github.com/odoo/odoo/pull/167751`
- `https://github.com/odoo/odoo/pull/170977`
- `https://github.com/odoo/odoo/pull/180999`

File: document_page/views/document_page_category.xml [hunk 1]
```diff
@@ -109,132 +106,86 @@
...
+                                    class="o_field_image fa fa-folder-open"
+                                    style="font-size: 64px; color: lightslategray"
+                                />
+                            </t>
+                            <t t-if="record.type.raw_value == 'content'">
+                                <i
+                                    class="o_field_image fa fa-file"
+                                    style="font-size: 64px; color: lightgray"
+                                />
+                            </t>
```
Comment (thienvh332) [inline document_page/views/document_page_category.xml]: We shouldn't discriminate against record.type.raw_value == 'content' like that. Let his image be displayed.
```suggestion
<t t-if="record.type.raw_value == 'category'">
                                <field
                                        name="image"
                                        t-if="record.image.raw_value"
                                        widget="image"
                                        t-att-alt="record.display_name.raw_value"
                                    />
                                <i
                                        t-if="!record.image.raw_value"
                                        class="o_field_image fa fa-folder-open"
                                        style="font-size: 64px; color: lightslategray"
                                    />
                            </t>
                            <t t-if="record.type.raw_value == 'content'">
                                <field
                                        name="image"
                                        t-if="record.image.raw_value"
                                        widget="image"
                                        t-att-alt="record.display_name.raw_value"
                                    />
                                <i
                                    t-if="!record.image.raw_value"
                                    class="o_field_image fa fa-file"
                                    style="font-size: 64px; color: lightgray"
                                />
                            </t>
```

File: document_page/static/src/js/document_page_kanban.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                this.modelName === "document.page" &&
+                this.$(".o_document_page_kanban_boxes a").length
+            ) {
+                this.$(".o_document_page_kanban_boxes a").first().click();
+            } else {
+                this._super.apply(this, arguments);
+            }
+        },
+    });
+});
```
Comment (thienvh332) [inline document_page/static/src/js/document_page_kanban.js]: How about this? From odoo 16, 17 we have switched to using **`OWL framework`** for odoo js. Please switch it back.

**Review by thienvh332 — APPROVED** [review]
LGTM

**Pull request closed without merging**